src/boot/boot.c

```diff
@@ -2950,8 +2956,12 @@ static EFI_STATUS run(EFI_HANDLE image) {
                 (void) process_random_seed(root_dir);
 
                 err = image_start(image, entry);
-                if (err != EFI_SUCCESS)
-                        return err;
+                if (err != EFI_SUCCESS) {
+                        if (config.reboot_on_err)
+                                reboot_system();
```

Comment:
before rebooting we should print a message explaining the situation, and then sleep for 5s or so (and saying so in the message), so that users can see what's going on. otherwise their screen would be flushed immediately quite likely with no chance to debug this.